File: C++/Problems/Problems/problem_1685.cpp
#include "problem_1685.h"

#include <cmath>

bool problem_1685::test() {
    std::vector nums { 1, 4, 6, 8, 10 };

    std::vector expected { 24, 15, 13, 15, 21 };

    auto result = getSumAbsoluteDifferences(nums);

    print1DVector(result);

    return expected == result;
}

std::vector<int> problem_1685::getSumAbsoluteDifferences(std::vector<int> &nums) {
    int n = nums.size();
    std::vector<int> result(n);
    int sum = 0;
    for (auto num : nums)
        sum += num;
    int preSum = 0;
    for (int i = 0; i < n; ++i)
    {
        int sumAfter = sum - preSum - (n - i) * nums[i];
        int sumBefore = i * nums[i] - preSum;
        result[i] = sumAfter + sumBefore;
        preSum += nums[i];
    }
    return result;
}

File: Python/Problems/Problems/problem1685.py
import heapq
import math
import sys

import numpy as np

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter

from .problem import Problem

class Problem1685(Problem):
    def test(self) -> bool:
        nums = [1, 4, 6, 8, 10]

        expected = [24, 15, 13, 15, 21]

        result = self.getSumAbsoluteDifferences(nums)

        print1DArray(result)

        return expected == result

    def getSumAbsoluteDifferences(self, nums: List[int]) -> List[int]:
        n = len(nums)
        result = []
        s = sum(nums)

        preSum = 0
        for i in range(n):
            sumAfter = s - preSum - (n - i) * nums[i]
            sumBefore = i * nums[i] - preSum
            result.append(sumAfter + sumBefore)
            preSum += nums[i]
        return result